MoreLinq.Test/UnfoldTest.cs

```diff
@@ -0,0 +1,120 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System.Linq;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class UnfoldTest
+    {
+        [Test]
+        public void UnfoldGenericWithNullGenerator()
+        {
+            Assert.ThrowsArgumentNullException("generator", () =>
+                MoreEnumerable.Unfold<int, Tuple<int, int>, int>(0, null, _ => true, e => e.First, e => e.Second));
+        }
+
+        [Test]
+        public void UnfoldGenericWithNullPredicate()
+        {
+            Assert.ThrowsArgumentNullException("predicate", () =>
+                MoreEnumerable.Unfold(0, e => new Tuple<int, int>(e, e + 1), null, e => e.First, e => e.Second));
```

Comment:
Use [`Tuple.Create`](https://msdn.microsoft.com/en-us/library/system.tuple.create.aspx) instead of `new Tuple<int, int>` for readability. Ditto elsewhere.